robolectric-shadows/shadows-support-v4/src/main/java/org/robolectric/shadows/support/v4/ShadowDrawerLayout.java

```diff
@@ -14,15 +17,36 @@
 @Implements(DrawerLayout.class)
 public class ShadowDrawerLayout extends ShadowViewGroup {
   @RealObject private DrawerLayout realDrawerLayout;
-  private DrawerLayout.DrawerListener drawerListener;
+  private final List<DrawerLayout.DrawerListener> drawerListeners = new ArrayList<>();
 
+  @Deprecated
   @Implementation
   public void setDrawerListener(DrawerLayout.DrawerListener drawerListener) {
-    this.drawerListener = drawerListener;
+    this.drawerListeners.clear();
+    this.drawerListeners.add(drawerListener);
     directlyOn(realDrawerLayout, DrawerLayout.class).setDrawerListener(drawerListener);
   }
 
+  @Implementation
+  public void addDrawerListener(DrawerLayout.DrawerListener drawerListener) {
+    this.drawerListeners.add(drawerListener);
+    directlyOn(realDrawerLayout, DrawerLayout.class).addDrawerListener(drawerListener);
+  }
+
+  @Implementation
+  public void removeDrawerListener(DrawerLayout.DrawerListener drawerListener) {
+    if (drawerListener != null) {
+      this.drawerListeners.remove(drawerListener);
+      directlyOn(realDrawerLayout, DrawerLayout.class).removeDrawerListener(drawerListener);
+    }
+  }
+
+  @Deprecated
   public DrawerLayout.DrawerListener getDrawerListener() {
```

Comment:
I think it might be preferable that instead of reimplementing / capturing the drawerlisteners we just dip into the implementation and return a copy of the list, its private so we'd need to get it by reflection though.

Personally, I'm not sure I even like exposing the listeners registered as I'd rather test behavior than check the listeners or fire them manually when I think they can be fired using DrawerListener.onViewDragStateChanged() - what do you think?